app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala

```diff
@@ -0,0 +1,165 @@
+    lazy val isSoleTraderBusiness = user.incomeSources.businesses.exists(business =>
+      business.incomeSourceId.contains(id) && !business.isCeased)
+
+    lazy val isForeignProperty = user.incomeSources.properties.exists(property =>
+      property.incomeSourceId.contains(id) && property.isForeignProperty && !property.isCeased)
+
+    lazy val isUKProperty = user.incomeSources.properties.exists(property =>
+      property.incomeSourceId.contains(id) && property.isUkProperty && !property.isCeased)
+
+    Future(
+      (isEnabled(IncomeSources), getTaxYearStartYearEndYear(taxYear), getReportingMethodKey(changeTo)) match {
+        case (false, _, _) =>
+          Ok(
+            customNotFoundErrorView()
+          )
+        case (_, _, None) =>
+          Logger("application")
+            .error(s"[ConfirmReportingMethodSharedController][handleRequest]: Could not parse reporting method: $changeTo")
+          itvcErrorHandler.showInternalServerError()
+        case (_, None, _) =>
+          Logger("application")
+            .error(s"[ConfirmReportingMethodSharedController][handleRequest]: Could not parse taxYear: $taxYear")
+          itvcErrorHandler.showInternalServerError()
+        case (_, Some(taxYear), Some(reportingMethodKey)) if isUKProperty =>
+          Ok(
+            confirmUKPropertyReportingMethod(
+              isAgent = isAgent,
+              taxYearEndYear = taxYear.endYear,
+              taxYearStartYear = taxYear.startYear,
+              reportingMethodKey = reportingMethodKey
+            )
+          )
+        case (_, Some(taxYear), Some(reportingMethodKey)) if isForeignProperty =>
+          Ok(
+            confirmForeignPropertyReportingMethod(
+              isAgent = isAgent,
+              taxYearEndYear = taxYear.endYear,
+              taxYearStartYear = taxYear.startYear,
+              reportingMethodKey = reportingMethodKey
+            )
+          )
+        case (_, Some(taxYear), Some(reportingMethodKey)) if isSoleTraderBusiness =>
+          Ok(
+            confirmSEReportingMethod(
+              isAgent = isAgent,
+              taxYearEndYear = taxYear.endYear,
+              taxYearStartYear = taxYear.startYear,
+              reportingMethodKey = reportingMethodKey
+            )
+          )
+        case _ =>
+          Logger("application")
+            .error(s"[ConfirmReportingMethodSharedController][handleRequest]: Could not find property or business with incomeSourceId: $id")
+          itvcErrorHandler.showInternalServerError()
+      }
+    )
+  }
+
+  private def getTaxYearStartYearEndYear(year: String): Option[TaxYear] = {
+
+    def isValidYear(year: String) = year.length == 4 && year.forall(_.isDigit)
+
+    year.split('-') match {
+      case Array(yearOne, yearTwo) if isValidYear(yearOne) && isValidYear(yearTwo) =>
+        Some(TaxYear(yearOne, yearTwo))
+      case _ => None
+    }
+  }
+
+  private def getReportingMethodKey(reportingMethod: String): Option[String] = {
```

Comment:
not sure why we need this method in term of the usage;
also its trivial duplication of the next logic:
  
Set("annual", "quarterly").find(_.toLowerCase == "w")